### ISSUE TITLE
m4c_captioner val-run: TypeError: unsupported operand type(s) for -=: 'NoneType' and 'int'

### ISSUE THREAD

**Issue by jc138691** [facebookresearch/mmf#847]
If you do not know the root cause of the problem, and wish someone to help you, please post according to this template:

## Instructions To Reproduce the Issue:

Check https://stackoverflow.com/help/minimal-reproducible-example for how to ask good questions. Simplify the steps to reproduce the issue using suggestions from the above link, and provide them below:

1. full code you wrote or full changes you made (`git diff`)
2. what exact command you run:

mmf_predict datasets=textcaps \
    model=m4c_captioner \
    config=projects/m4c_captioner/configs/m4c_captioner/textcaps/defaults.yaml \
    env.save_dir=./save/m4c_captioner/defaults \
    run_type=val \
    checkpoint.resume_zoo=m4c_captioner.textcaps.defaults

Dear mmf-team, I am trying out your example how to: Generate prediction file on the validation set, via
# https://mmf.sh/docs/projects/m4c_captioner/
# mmf_predict datasets=textcaps
# model=m4c_captioner
# config=projects/m4c_captioner/configs/m4c_captioner/textcaps/defaults.yaml
# env.save_dir=./save/m4c_captioner/defaults
# run_type=val
# checkpoint.resume_zoo=m4c_captioner.textcaps.defaults

3. __full logs__ you observed:
And getting the following error:

WARNING 2021-04-01T17:30:33 | mmf: Key textcaps_num_final_outputs is not present in registry, returning default value of File "/Users/jc138691/dev/comp/mmf_v210330/mmf/trainers/mmf_trainer.py", line 43, in load
super().load()
File "/Users/jc138691/dev/comp/mmf_v210330/mmf/trainers/base_trainer.py", line 30, in load
self.load_model()
File "/Users/jc138691/dev/comp/mmf_v210330/mmf/trainers/mmf_trainer.py", line 94, in load_model
self.model = build_model(attributes)
File "/Users/jc138691/dev/comp/mmf_v210330/mmf/utils/build.py", line 94, in build_model
model.build()
File "/Users/jc138691/dev/comp/mmf_v210330/mmf/models/m4c.py", line 45, in build
self._build_output()
File "/Users/jc138691/dev/comp/mmf_v210330/mmf/models/m4c.py", line 157, in _build_output
num_choices -= self.config.classifier.ocr_max_num
TypeError: unsupported operand type(s) for -=: 'NoneType' and 'int'

## Environment:
Provide your environment information using the following command:
```
python -m torch.utils.collect_env
python -m torch.utils.collect_env
Collecting environment information...
PyTorch version: 1.6.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Ubuntu 20.04.2 LTS
GCC version: (Ubuntu 7.5.0-6ubuntu2) 7.5.0
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 460.67
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.20.2
[pip3] pytorch-lightning==1.1.6
[pip3] torch==1.6.0
[pip3] torchtext==0.5.0
[pip3] torchvision==0.7.0
[conda] numpy                     1.20.2                   pypi_0    pypi
[conda] pytorch-lightning         1.1.6                    pypi_0    pypi
[conda] torch                     1.6.0                    pypi_0    pypi
[conda] torchtext                 0.5.0                    pypi_0    pypi
[conda] torchvision               0.7.0                    pypi_0    pypi
```

Everything was installed as per your: https://mmf.sh/docs/
git clone https://github.com/facebookresearch/mmf.git
cd mmf
pip install --editable .

**Comment by ronghanghu** [comment]
fixed in #851

**Comment by 2017210384** [comment]
I had the same problem

**Comment by ronghanghu** [comment]
@2017210384 This should have been addressed now. Could you try pulling the latest MMF after the fix in #851?

**Comment by 2017210384** [comment]
@ronghanghu Thanks.
